client/src/lib/CKANApi/DataStore.js

```diff
@@ -0,0 +1,62 @@
+import CKANApi from '../CKANApi';
+
+export default class {
+  constructor(endpoint, resource) {
+    this.endpoint = endpoint;
+    this.resource = resource;
+  }
+
+  /**
+   *
+   * @param {string[]} fields - The fields to return
+   * @param {string|object} term - A string term to search globally or an object of field : term
+   * @param {boolean} distinct
+   */
+  search(fields = [], term = null, distinct = false) {
+    if (!fields.length) {
+      return false;
+    }
+
+    const options = {
+      id: this.resource,
+      fields: fields.map(encodeURIComponent).join(','),
+    };
+
+    // If an invalid term is given then assume that it will return nothing
+    const termType = term && typeof term;
```

Comment:
I'm personally a bit against this kind of ternary assignment. I read it as "termType is the [boolean] evaluation of `term && typeof term`". Maybe `const termType = term ? typeof term : null;` might be easier to read?

Reply:
Fair. I will just update to early return a falsy value of `term` and then we can avoid the ternary assignment all together :)